config/config.go

```diff
@@ -50,6 +51,28 @@ func FromYAMLFile(name string, v Validator) error {
 	return nil
 }
 
+// EnvOptions is a type alias for [env.Options], so that only this package needs to import [env].
+type EnvOptions = env.Options
+
+// FromEnv parses environment variables and stores the result in the value pointed to by v.
+// If v is nil or not a pointer, FromEnv returns an [ErrInvalidArgument] error.
+func FromEnv(v Validator, options EnvOptions) error {
+	rv := reflect.ValueOf(v)
+	if rv.Kind() != reflect.Ptr || rv.IsNil() {
+		return errors.Wrapf(ErrInvalidArgument, "non-nil pointer expected, got %T", v)
+	}
+
+	if err := defaults.Set(v); err != nil {
+		return errors.Wrap(err, "can't set config defaults")
+	}
+
+	if err := env.ParseWithOptions(v, options); err != nil {
+		return errors.Wrap(err, "can't parse environment variables")
+	}
+
+	return errors.Wrap(v.Validate(), "invalid configuration")
```

Comment:
Personally, since I discovered that `errors.Wrap()` also handles nil errors, I do tend to use it as the last statement as well, instead of using an if statement and returning nil at the end. However, @oxzi also requested a change in https://github.com/Icinga/icinga-go-library/pull/63#discussion_r1717079220 for not using it for readability reasons, so if you think this makes the code more readable/understandable, we can make it the standard.

Reply:
In the referenced comment, I primarily wanted a consistence between the two checks. Furthermore, when I am reading `errors.Wrap(…)`, I am expecting an error to be handled and wrapped. Adding the if guard makes it more readable, imo. But honestly, I am fine with both.

Reply:
Ok, I'd like to define this now: Don't return `errors.Wrap()` directly if the error could be `nil`.